src/schemaCompare/schemaCompareWebViewController.ts

```diff
@@ -164,52 +169,80 @@ export class SchemaCompareWebViewController extends ReactWebviewPanelController<
         );
     }
 
+    /**
+     * Checks if the given node is a SQL project node with a valid project file path.
+     * Returns true if the node is not a TreeNodeInfo, has a projectFilePath property,
+     * and its treeDataProvider contains a root with a valid projectFileUri.fsPath.
+     */
+    private isSqlProjectNodeWithFilePath(node: any): boolean {
```

Comment:
This seems like it should be in the SQL Projects extension instead of MSSQL.  When the Project -> Schema Compare command is moved to the SqlProj extension, it will have the project path directly that can be passed from there.

Reply:
True, since we are handling the SC command from sqlproj side, sending the source/target as just sqlproj file path would remove the unnecessary checks. refactored the code accordingly.